tests/test_homogenize.py

```diff
@@ -175,3 +177,71 @@ def test_rasterize_vector_fail(candidate_map, benchmark_map, rasterize_attribute
             benchmark_map=benchmark_map,
             rasterize_attributes=rasterize_attributes,
         )
+
+
+@parametrize_with_cases(
+    "candidate_map, benchmark_map, expected", glob="numeric_align_dataarrays"
+)
+def test_numeric_align_dataarrays(candidate_map, benchmark_map, expected):
+    """Tests numeric alignment"""
+
+    c, b = _align_numeric_data_type(candidate_map, benchmark_map)
+
+    assert c.data.dtype == expected
+    assert b.data.dtype == expected
+
+    assert np.isclose(
```

Comment:
When converting from float32 to float64 and other conversions will induce a small change in the value since there is a conversion in precision.  For a mean this will compound these small errors within a certain tolerance, thus the check is to make sure that conversion preserves the original numbers within a certain tolerance, in this case 1e-5.